sdk/python/symbolchain/sc/__init__.py

```diff
@@ -662,18 +663,22 @@ def deserialize(cls, payload: bytes | bytearray | memoryview) -> Transaction:
 		deadline = Timestamp.deserialize(buffer)
 		buffer = buffer[deadline.size:]
 
-		instance = Transaction()
+		# pylint: disable=protected-access
 		instance._signature = signature
 		instance._signer_public_key = signer_public_key
 		instance._version = version
 		instance._network = network
 		instance._type_ = type_
 		instance._fee = fee
 		instance._deadline = deadline
-		return instance
+		return (size_ - len(buffer), size_)
```

Comment:
Should ``len(buffer)`` be zero at end? 🤔

Reply:
no - buffer is entire *remaining* buffer. so, we need to effectively shift (and mask) it like in the JS implementation:
```
		view.shiftRight(4);
		view.shrink(size - 4);
```
remember, this is base class (Transaction) and size includes both base + derived fields. at this point, buffer should contain only derived part. `size_ - len(buffer) ` let's us skip to derived part. `size_` masks off the next entity(ies), if present [important for aggregates, which have cosignatures expanding to fill remaining space].